src/support/mtx_rw.c

```diff
@@ -245,6 +243,8 @@ stall:			__wt_cond_wait(
 	 * meantime.
 	 */
 	WT_READ_BARRIER();
+
+	WT_ASSERT(session, l->u.s.readers_active > 0);
```

Comment:
@michaelcahill, I'm not seeing this one. The lock structure is volatile, so this read cannot be cached/re-ordered, and the thread of control is holding a read lock so `readers_active` must be non-zero?